presto-main/src/main/java/com/facebook/presto/operator/aggregation/KeyValuePairs.java

```diff
@@ -80,11 +93,79 @@ public Slice serialize()
     {
         Block values = valueBlockBuilder.build();
         Block keys = keyBlockBuilder.build();
-        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), keys.getSizeInBytes() + values.getSizeInBytes());
+        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), keys.getSizeInBytes() + values.getSizeInBytes() + BooleanType.BOOLEAN.getFixedSize() * 3);
+
+        BooleanType.BOOLEAN.writeBoolean(blockBuilder, isMultiValue);
```

Comment:
I think serializing this boolean will become a problem when you rebase on https://github.com/facebook/presto/pull/3226
Instead I would add a parameter (`isMultiMap`) to `KeyValuePairStateSerializer` and pass that through into the constructor. That also means that you can make `KeyValuePairs.isMultiValue` a final field